lib/ansible/template/__init__.py

```diff
@@ -446,17 +625,18 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
 
                         unsafe = hasattr(result, '__UNSAFE__')
                         if convert_data and not self._no_type_regex.match(variable):
-                            # if this looks like a dictionary or list, convert it to such using the safe_eval method
-                            if (result.startswith("{") and not result.startswith(self.environment.variable_start_string)) or \
-                                    result.startswith("[") or result in ("True", "False"):
-                                eval_results = safe_eval(result, locals=self._available_variables, include_exceptions=True)
-                                if eval_results[1] is None:
-                                    result = eval_results[0]
-                                    if unsafe:
-                                        result = wrap_var(result)
-                                else:
-                                    # FIXME: if the safe_eval raised an error, should we do something with it?
-                                    pass
+                            if isinstance(result, (str, unicode, bytes)):
```

Comment:
this is not compatible across python versions use text_strings, byte_strings from our 'compatiblity' code instead